### PR TITLE
document manual steps to update launch_2 and launch_heavy_1 keyboards

### PR DESCRIPTION
**Pull request by thomas-zimmerman**
create online guide for manual steps to update firmware on these keyboards.

### CODE REVIEW

**Review by ahoneybun — APPROVED** [review]
I think the article title should just be launch or similar since it also goes over the launch_heavy_1 so it does not confuse customers.

**Review by 13r0ck — DISMISSED** [review]
Adding my review too as I was lightly involved. Thanks again Thomas!

File: content/launch_2-firmware-update.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+```
+sudo apt remove test-launch-system76-keyboard-configurator
+sudo apt autoremove
+```
+
+Future updates will now be done with `fwupd` and should not require these manual steps.
+
+## Getting help
+
+If this does not work, please open a support case on for your Keyboard. Factory updates to firmware done via ISP cable can be arranged.
```
Comment (leviport) [inline content/launch_2-firmware-update.md:70]: s/Keyboard/keyboard

File: content/launch_2-firmware-update.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+### Install software
+
+We have a package that will bring in the needed support software to do this firmware update. Run the following command in a terminal:
+
+```
+sudo apt install test-launch-system76-keyboard-configurator
+```
+
+If you are not on Pop!\_OS you can also use the AppImage version of the `system76-keyboard` and luanch from the command line and add the option `--lanch-test`. You will also need to make sure to install `avrdude` and `dfu-programmer`.
```
Comment (leviport) [inline content/launch_2-firmware-update.md:30]: s/`--lanch-test`/`--launch-test`

Comment (leviport) [inline content/launch_2-firmware-update.md:30]: >If you are not on Pop!\_OS you can also use the AppImage

Should probably be "Pop!\_OS or Ubuntu"

Comment (leviport) [reply content/launch_2-firmware-update.md:30]: >version of the `system76-keyboard` and

I think this should be `system76-keyboard-configurator`

Comment (leviport) [reply content/launch_2-firmware-update.md:30]: >You will also need to make sure to install `avrdude` and `dfu-programmer`.

"to make sure" can be removed, simplifying this sentence to:
>You will also need to install `avrdude` and `dfu-programmer`.

**Review by leviport — CHANGES_REQUESTED** [review]

File: content/launch_2-firmware-update.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+facebookImage: /_social/article
+twitterImage: /_social/article
+
+hidden: false
+section: software-troubleshooting
+tableOfContents: true
+---
+
+## Steps to take to update firmware
```
Comment (leviport) [inline content/launch_2-firmware-update.md:20]: I think starting this with a summary of which keyboard versions this applies to (launch_2 and launch_heavy_1), and in which circumstances this article will be necessary (when prompted to update, but the update doesn't install).

File: content/launch_2-firmware-update.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+If you are not on Pop!\_OS you can also use the AppImage version of the `system76-keyboard` and luanch from the command line and add the option `--lanch-test`. You will also need to make sure to install `avrdude` and `dfu-programmer`.
+
+Start System76 Keyboard Tester -- this will launch the `System76 Keyboard Configurator` in a test mode
+
+![System76-Tester](/images/launch_2-firmware-update/Launch-system76-keyboard-tester.png)
+
+### Enter Bootloader mode
+
+Unplug the launch keyboard and hold <kbd>ESC</kbd> while plugging it back in.
```
[the remark this answers is not in the document]
Comment (thomas-zimmerman) [reply content/launch_2-firmware-update.md:38]: Continue to hold the <kbd>ESC</kbd> key until you see the firmware update button appear in the top right of the `System76 Keyboard Configurator` or 10 seconds has passed.

File: content/launch_2-firmware-update.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomas-zimmerman) [reply content/launch_2-firmware-update.md:30]: Done.

File: content/launch_2-firmware-update.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+*NOTE: This may fail. Try 5 times -- in-between retries, leave the keyboard unplugged and press no buttons for 15 seconds.
+![System76-Tester](/images/launch_2-firmware-update/Tester-in-firmware-update-mode.png)
+
+### Suscessful enter of firmware update mode
```
[the remark this answers is not in the document]
Comment (thomas-zimmerman) [reply content/launch_2-firmware-update.md:43]: Pulling this out of a note and into the paragraph

File: content/launch_2-firmware-update.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+You will know you have succedded when a new button shows up in the header of the keyboard configurator. This button will list firmware available for your keyboard. Select the "launch_2" or "lauanch_heavy_1" based on the keyboard you have. A successful firmware update will freeze updates to the application as the firmware is flashed to the new firmware.
```
[the remark this answers is not in the document]
Comment (thomas-zimmerman) [reply content/launch_2-firmware-update.md:45]: edited this section -- it should read a bit better now. Warning added.

File: content/launch_2-firmware-update.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+Use `fwupdmgr` to verify that the firmware has updated to at least `0.12.20`:
+
+```
+sudo fwupdmgr get-devices
+```
+![Firmware version](/images/launch_2-firmware-update/launch-version-number.png)
+
+### Clean up and remove software
+
+We can remove the un-needed software with:
```
[the remark this answers is not in the document]
Comment (thomas-zimmerman) [reply content/launch_2-firmware-update.md:59]: Changed the wording to not need "unneeded".

**Comment by leviport** [comment]
I pushed some updates that should address most of the comments I made. I still need a screenshot for the "Verify firmware update" section if we want to instruct people to check the Firmware Manager for updates rather than checking `fwupd` output. Let me know what you think about that one.

File: content/launch_2-firmware-update.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+![System76-Tester](/images/launch_2-firmware-update/Tester-in-firmware-update-mode.png)
+
+### Suscessful enter of firmware update mode
+
+You will know you have succedded when a new button shows up in the header of the keyboard configurator. This button will list firmware available for your keyboard. Select the "launch_2" or "lauanch_heavy_1" based on the keyboard you have. A successful firmware update will freeze updates to the application as the firmware is flashed to the new firmware.
+![System76-Tester-firmware-update](/images/launch_2-firmware-update/Tester-in-firmware-update-mode-menu.png)
+
+### Verify firmware update
+
+Use `fwupdmgr` to verify that the firmware has updated to at least `0.12.20`:
```
[the remark this answers is not in the document]
Comment (leviport) [reply content/launch_2-firmware-update.md]: Removed this section entirely

**Review by leviport — DISMISSED** [review]
I like it, but a lot of it is my work, so more eyes is probably a good idea.

**Comment by thomas-zimmerman** [comment]
I think this looks really good -- I would mark this as reviewed but as the author of the original PR, I can't.

Looking over this live, all of the images are fine.

File: content/launch_2-firmware-update.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+If you are not on Pop!\_OS or Ubuntu, you can also use the AppImage version of the System76 Keyboard Configurator and launch it with the `--launch-test` command line argument. You will also need to install `avrdude` and `dfu-programmer` if they are not installed already.
+
+Start System76 Keyboard Tester -- this will launch the System76 Keyboard Configurator in a test mode.
+
+![System76-Tester](/images/launch_2-firmware-update/Launch-system76-keyboard-tester.png)
+
+### Enter Bootloader mode
+
+Unplug the launch keyboard and hold <kbd>Esc</kbd> while plugging it back in. Continue to hold the <kbd>Esc</kbd> key for about 10 seconds, or until the firmware update button appears in the right side of the Configurator's header bar. The keyboard's LEDs will remain off while the keyboard is in bootloader mode.
```
Comment (n3m0-22) [inline content/launch_2-firmware-update.md:46]: I believe in "Unplug the launch keyboard" Launch Keyboard should be capitalized.

**Review by n3m0-22 — CHANGES_REQUESTED** [review]

File: content/launch_2-firmware-update.md (hunk 9 above)
Comment (leviport) [reply content/launch_2-firmware-update.md]: Fixed with [0c840a2](https://github.com/system76/docs/pull/1140/commits/0c840a2689c4bb874b121306f7241aa6c1d48363)

**Review by n3m0-22 — APPROVED** [review]

**Comment by leviport** [comment]
This shouldn't be merged until the configurator update is released: https://github.com/pop-os/repo-release/pull/249

**Pull request merged**